doc/ref/gappkg.xml

```diff
@@ -173,5 +188,1996 @@ users.
+<C>.g</C>, <C>.gd</C> and <C>.gi</C> files (other than <F>PackageInfo.g</F>,
+<F>init.g</F> and <F>read.g</F>). For some packages, the directory <F>gap</F>
+has been used instead of <F>lib</F>; <F>lib</F> has the advantage that it is
+the default subdirectory of a package directory searched for by the
+<Ref Func="DirectoriesPackageLibrary"/> command.
+</Item>
+
+<Mark><F>src</F></Mark>
+<Item>
+If the package contains non-&GAP; code, e.g.&nbsp;C code, then this source
+code should go in the <F>src</F> directory. If there are <C>.h</C>
+<Q>include</Q> files you may prefer to put these all together in a separate
+<C>include</C> directory. There is one further rule for the location of kernel
+library modules or external programs which is explained in 
+<Ref Sect="Installation of GAP Package Binaries"/> below.
+</Item>
+
+<Mark><F>tst</F></Mark>
+<Item>
+It is highly recommended that a package should have test files, which then 
+should go in the <F>tst</F> directory. For a deposited package, a test file 
+with a basic test of the package (for example, to check that it works as 
+expected and/or that the manual examples are correct) may be specified in the 
+<File>PackageInfo.g</File> to be included in the &GAP; standard test suite
+and run as a part of the &GAP; release preparation.
+More specific and time consuming tests are not supposed to be a part of the 
+&GAP; standard test suite but may be placed in the <F>tst</F> directory
+with further instructions on how to run them.
+See Section <Ref Sect="Testing a GAP package"/> about the requirements
+to the test files formats and further recommendations.
+</Item>
+
+</List>
+
+All other files can be organised as you like. But we suggest that you
+have a look at existing packages and use a similar scheme, for
+example, put examples in the <File>examples</File> subdirectory, data 
+libraries in extra subdirectories, and so on.
+<P/>
+Sometimes there may be a need to include an empty directory in the
+package distribution (for example, as a place to store some data that 
+may appear at runtime). In this case package authors are
+advised to put in this directory a short <F>README</F> file describing
+its purpose to ensure that such directory will be included in the 
+redistribution.
+<P/>
+Concerning the &GAP; code in packages, it is recommended to use only
+documented &GAP; functions, see <Ref Sect="Undocumented Variables"/>.
+In particular if you want to make your package available to other &GAP; users
+it is advisable to avoid using <Q>obsolete</Q> variables
+(see <Ref Chap="Replaced and Removed Command Names"/>). To test that
+the package does not use obsolete variables you can set the <C>ReadObsolete</C> 
+component in your <F>gap.ini</F> file to <K>false</K> (see
+<Ref Sect="sect:gap.ini"/>) or start &GAP; with <C>-O</C> command line
+option (this may also cause problems with loading other packages using
+<Q>obsolete</Q> variables).
+
```

Comment:
`gap -O` might be a bit discouraging; probably `gap -A -O` would be better, since even many default autoloaded packages contain obsolete functions

Reply:
@pedritomelenas thanks, good suggestion!